File: samples/CSharp/Northwind/Order.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Orders")]
   public class Order {

      [Column(IsPrimaryKey = true, IsDbGenerated = true)]
      public int OrderID { get; set; }

      [Column]
      public string CustomerID { get; set; }

      [Column]
      public int? EmployeeID { get; set; }

      [Column]
      public DateTime? OrderDate { get; set; }

      [Column]
      public DateTime? RequiredDate { get; set; }

      [Column]
      public DateTime? ShippedDate { get; set; }

      [Column]
      public int? ShipVia { get; set; }

      [Column]
      public decimal? Freight { get; set; }

      [Column]
      public string ShipName { get; set; }

      [Column]
      public string ShipAddress { get; set; }

      [Column]
      public string ShipCity { get; set; }

      [Column]
      public string ShipRegion { get; set; }

      [Column]
      public string ShipPostalCode { get; set; }

      [Column]
      public string ShipCountry { get; set; }

      [Association(OtherKey = nameof(OrderDetail.OrderID))]
      public Collection<OrderDetail> OrderDetails { get; } = new Collection<OrderDetail>();

      [Association(ThisKey = nameof(CustomerID))]
      public Customer Customer { get; set; }

      [Association(ThisKey = nameof(EmployeeID))]
      public Employee Employee { get; set; }

      [Association(ThisKey = nameof(ShipVia))]
      public Shipper Shipper { get; set; }
   }
}

File: samples/VisualBasic/Northwind/Order.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Orders")>
   Public Class Order

      <Column>
      Property CustomerID As String

      <Column>
      Property EmployeeID As Integer?

      <Column>
      Property Freight As Decimal?

      <Column>
      Property OrderDate As Date?

      <Column(IsPrimaryKey:=True, IsDbGenerated:=True)>
      Property OrderID As Integer

      <Column>
      Property RequiredDate As Date?

      <Column>
      Property ShipAddress As String

      <Column>
      Property ShipCity As String

      <Column>
      Property ShipCountry As String

      <Column>
      Property ShipName As String

      <Column>
      Property ShippedDate As Date?

      <Column>
      Property ShipPostalCode As String

      <Column>
      Property ShipRegion As String

      <Column>
      Property ShipVia As Integer?

      <Association(ThisKey:=NameOf(CustomerID))>
      Property Customer As Customer

      <Association(ThisKey:=NameOf(EmployeeID))>
      Property Employee As Employee

      <Association(OtherKey:=NameOf(OrderDetail.OrderID))>
      ReadOnly Property OrderDetails As New Collection(Of OrderDetail)

      <Association(ThisKey:=NameOf(ShipVia))>
      Property Shipper As Shipper

   End Class

End Namespace
